src/core/ddsc/src/dds_write.c

```diff
@@ -115,6 +121,36 @@ dds_return_t dds_loan_sample(dds_entity_t writer, void** sample)
 #endif
 }
 
+dds_return_t dds_return_writer_loan(dds_writer *writer, void *sample)
+{
+#ifndef DDS_HAS_SHM
+  (void) writer;
+  (void) sample;
+  return DDS_RETCODE_UNSUPPORTED;
+#else
+  dds_return_t ret;
+
+  if (!sample)
```

Comment:
This being not an external API, I think it makes more sense to require `sample` to be non-null.

Reply:
Hi @eboasson, thanks for your review. I processed your review comments in a subsequent commit, so you can clearly see the diffs I made. If you agree, could you squash commit this feature into the master?

Reply:
BTW: There is still one failing testcase, but that seems to be unconnected to this Pull Request. It is a testcase that test security plugins, and it complains about an expired CRL. Maybe you or @k0ekk0ek can take care of that?